eventcore-postgres/src/lib.rs

```diff
@@ -0,0 +1,292 @@
+use std::time::Duration;
+
+use eventcore::{
+    Event, EventStore, EventStoreError, EventStreamReader, EventStreamSlice, StreamId,
+    StreamWriteEntry, StreamWrites,
+};
+use serde_json::{Value, json};
+use sqlx::types::Json;
+use sqlx::{Pool, Postgres, Row, postgres::PgPoolOptions, query, query_scalar};
+use std::collections::HashMap;
+use thiserror::Error;
+use tracing::{info, instrument, warn};
+use uuid::Uuid;
+
+#[derive(Debug, Error)]
+pub enum PostgresEventStoreError {
+    #[error("failed to create postgres connection pool")]
+    ConnectionFailed(#[source] sqlx::Error),
+    #[error("failed to ping postgres")]
+    PingFailed(#[source] sqlx::Error),
+    #[error("failed to apply postgres migrations")]
+    MigrationFailed(#[source] sqlx::migrate::MigrateError),
+}
+
+#[derive(Debug, Clone)]
+pub struct PostgresEventStore {
+    pool: Pool<Postgres>,
+}
+
+impl PostgresEventStore {
+    pub async fn new<S: Into<String>>(
+        connection_string: S,
+    ) -> Result<Self, PostgresEventStoreError> {
+        let connection_string = connection_string.into();
+        let pool = PgPoolOptions::new()
+            .max_connections(5)
+            .acquire_timeout(Duration::from_secs(5))
+            .connect(&connection_string)
+            .await
+            .map_err(PostgresEventStoreError::ConnectionFailed)?;
+        Ok(Self { pool })
+    }
+
+    /// Test-only constructor that exists solely for mutation testing and integration harnesses.
+    pub fn from_pool_for_tests(pool: Pool<Postgres>) -> Self {
+        Self { pool }
+    }
```

Comment:
Yuck?

Reply:
Replaced with general-purpose `from_pool()` that's not test-specific. Users who want full control over pool configuration (or want to share a pool) can use this. Commit: 6831640